tests/unit/test_asymencryption.cpp

```diff
@@ -0,0 +1,1004 @@
+        {
+            0x40, 0x6B, 0x15, 0x44, 0xDF, 0x62, 0xCD, 0x33, 0x80, 0x13, 0x24, 0xFC, 0xC2, 0xAE,
+            0x18, 0x4D, 0x2F, 0xD6, 0x32, 0x16, 0xA3, 0x86, 0x1F, 0x66, 0x15, 0xF8, 0x09, 0xE0,
+            0xAB, 0x1E, 0x56, 0xC0, 0x4E, 0x35, 0x37, 0x11, 0x82, 0x07, 0xBD, 0xBF, 0xC8, 0x68,
+            0xBF, 0xF6, 0xF2, 0x61, 0xCC, 0x71, 0xBC, 0x38, 0x5A, 0x88, 0x1D, 0xFB, 0x1D, 0x98,
+            0x97, 0xAE, 0xBE, 0xF8, 0xE3, 0xDB, 0x02, 0x05, 0x8C, 0xE9, 0x16, 0x16, 0x32, 0x7E,
+            0x8D, 0x99, 0x4F, 0x85, 0x63, 0x8B, 0x36, 0xA0, 0xBD, 0xD0, 0xA4, 0x99, 0xA8, 0x01,
+            0x22, 0xB4, 0x21, 0xD3, 0x0B, 0xC6, 0x63, 0xE0, 0xA9, 0x4C, 0xA8, 0x7F, 0xB3, 0x36,
+            0x47, 0x64, 0x2F, 0x25, 0xB3, 0xF9, 0xBC, 0x51, 0xA7, 0x7D, 0x91, 0x4F, 0xA6, 0xDE,
+            0x57, 0x52, 0xF9, 0x24, 0x9C, 0x18, 0x76, 0xFC, 0xE4, 0xC0, 0x86, 0x67, 0x30, 0xA6,
+            0x60, 0x1B
+        },
+        keyPairs1024bit._publicKey,
+        keyPairs1024bit._privateKey,
+        openssl::RSAPaddingMode::OAEP,
+        openssl::DigestTypes::SHA256,
+        openssl::DigestTypes::SHA256,
+        {}
+    };
+
+    // Step 1: Check the decryption with a wrong padding mode.
+    const auto privateKey  = mococrw::AsymmetricKeypair::readPrivateKeyFromPEM(data._privateKey, "");
+    const auto wrongPaddingMode = PKCSPadding();
+    EXPECT_THROW(AsymmetricEncryption::decrypt(privateKey, wrongPaddingMode, data._encrypted),
+                 MoCOCrWException);
+
+    // Step 2: Check the decryption with a wrong hashing
+    const auto wrongHashing = OAEPPadding(openssl::DigestTypes::SHA512, data._masking, data._label);
+    EXPECT_THROW(AsymmetricEncryption::decrypt(privateKey, wrongHashing, data._encrypted),
+                 MoCOCrWException);
+
+    // Step 3: Check the decryption with a wrong masking
+    const auto wrongMasking = OAEPPadding(data._hashing, openssl::DigestTypes::SHA512, data._label);
+    EXPECT_THROW(AsymmetricEncryption::decrypt(privateKey, wrongMasking, data._encrypted),
+                 MoCOCrWException);
+
+    // Step 4: Check the decryption with a wrong label
+    const auto wrongLabel = OAEPPadding(data._hashing, data._masking, {0xAA, 0xBB, 0xCC, 0xDD});
+    EXPECT_THROW(AsymmetricEncryption::decrypt(privateKey, wrongLabel, data._encrypted),
+                 MoCOCrWException);
+
+    // Step 5: Check the decryption with a wrong key
+    const auto wrongPrivateKey
+            = mococrw::AsymmetricKeypair::readPrivateKeyFromPEM(keyPairs2048bit._privateKey, "");
+    const auto paddingMode = OAEPPadding(data._hashing, data._masking, data._label);
+    EXPECT_THROW(AsymmetricEncryption::decrypt(wrongPrivateKey, paddingMode, data._encrypted),
+                 MoCOCrWException);
+}
+
+/**
+ * @brief Tests the CryptoData class
+ * The following use-cases are covered:
+ * - constructors
+ * - assignment operators
+ * - operator<<
+ */
+TEST_F(AsymmetricEncryptionTest, testCryptoData)
```

Comment:
It may be moved into a separate test file as it has nothing to do with symmetric encryption.